### PR TITLE
Fixed #25307 -- conditional aggregation over subqueries

### PR DESCRIPTION
**Pull request by jarshwah**
A bug fix for Anssi's previous patch. Thanks to @jproffitt for identifying the source of the bug!

From original PR https://github.com/django/django/pull/6267/

### CODE REVIEW

**Comment by jarshwah** [comment]
buildbot, test on oracle.

**Comment by jarshwah** [comment]
buildbot, test on xenial.

File: django/db/models/sql/where.py [hunk 1]
```diff
@@ -118,6 +118,16 @@ def get_group_by_cols(self):
             cols.extend(child.get_group_by_cols())
         return cols
 
+    def get_source_expressions(self):
+        sources = []
+        for child in self.children:
+            sources.append(child)
+        return sources
```
Comment (charettes) [inline django/db/models/sql/where.py]: I believe you can shorten that to `return self.children[:]`.

File: django/db/models/sql/where.py [hunk 2]
```diff
@@ -160,6 +170,13 @@ def _contains_aggregate(cls, obj):
     def contains_aggregate(self):
         return self._contains_aggregate(self)
 
+    @property
+    def is_summary(self):
+        for obj in self.children:
+            if obj.is_summary:
+                return True
+        return False
```
Comment (charettes) [inline django/db/models/sql/where.py]: `return any(child.is_summary for child in self.children)`?

**Pull request merged**